src/internal/kopia/wrapper.go

```diff
@@ -515,3 +518,118 @@ func isErrEntryNotFound(err error) bool {
 	return strings.Contains(err.Error(), "entry not found") &&
 		!strings.Contains(err.Error(), "parent is not a directory")
 }
+
+func (w Wrapper) Maintenance(
+	ctx context.Context,
+	safety control.Safety,
+	quickMaintenance, force bool,
+) error {
+	kopiaSafety, err := translateSafety(safety)
+	if err != nil {
+		return clues.Wrap(err, "getting safety level")
+	}
+
+	mode := translateMode(quickMaintenance)
+	currentOwner := w.c.ClientOptions().UsernameAtHost()
+
+	ctx = clues.Add(
+		ctx,
+		"kopia_safety", kopiaSafety,
+		"kopia_maintenance_mode", mode,
+		"force", force,
+		"current_owner", clues.Hide(currentOwner))
+
+	dr, ok := w.c.Repository.(repo.DirectRepository)
+	if !ok {
+		return clues.New("unable to get valid handle to repo").WithClues(ctx)
+	}
+
+	// Below write session options pulled from kopia's CLI code that runs
+	// maintenance.
+	err = repo.DirectWriteSession(
+		ctx,
+		dr,
+		repo.WriteSessionOptions{
+			Purpose: "Corso maintenance",
+		},
+		func(ctx context.Context, dw repo.DirectRepositoryWriter) error {
```

Comment:
nit, could this get moved into a var declaration above this?  Imo, declaring funcs within a func call isn't as readable as the following.  (plus, reduces indent depth, and makes it easier to declare if we ever decide to not make it an ad-hoc func for testing purposes)
```go
ctx = ...
dr, ok := ...
if !ok {...}

theFunc := func(...)

err = repo.DirectWriteSession(ctx, dr, {}, theFunc)
```